themes/default/content/docs/intro/concepts/inputs-outputs.md

```diff
@@ -651,6 +651,339 @@ variables:
+        pulumi.ToMapOutput(map[string]pulumi.Output{
+            "key": pulumi.ToOutput("value"),
+        }),
+    }),
+    "map": pulumi.ToMapOutput(map[string]pulumi.Output{
+        "key": pulumi.ToOutput("value"),
+    }),
+    // The following functionality is currently unsupported as myResource
+    // is an unknown value.
+    "unknown": myResource.ApplyT(func(res interface{}) (interface{}, error) {
+        return "Hello World!", nil
+    }),
+}))
+```
+
+{{% /choosable %}}
+{{% choosable language csharp %}}
+
+```csharp
+var stateMachine = Pulumi.Output.JsonSerialize(Output.Create(new {
+        Comment = "A Hello World example of the Amazon States Language using two AWS Lambda Functions",
+        StartAt = "Hello",
+        States = new Dictionary<string, object?>{
+        ["Hello"] = new {
+            Type = "Task",
+            Resource = helloFunction.Arn, // Pulumi Resource Output
+            Next = "World",
+        },
+        ["World"] = new {
+            Type = "Task",
+            Resource = worldFunction.Arn, // Pulumi Resource Output
+            End = true,
+        },
+    },
+}));
+```
+
+{{% /choosable %}}
+{{< /chooser >}}
+
+### Example Parsing JSON Outputs to Objects
+
+The following is an example of Pulumi code that operates on a JSON object and removes all of the policy statements by replacing it with an empty list.
+
+{{< chooser language "javascript,typescript,python,go,csharp" >}}
+{{% choosable language javascript %}}
+
+```javascript
+const jsonIAMPolicy = pulumi.output(`{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Sid": "VisualEditor0",
+            "Effect": "Allow",
+            "Action": [
+                "s3:ListAllMyBuckets",
+                "s3:GetBucketLocation"
+            ],
+            "Resource": "*"
+        },
+        {
+            "Sid": "VisualEditor1",
+            "Effect": "Allow",
+            "Action": "s3:*",
+            "Resource": "arn:aws:s3:::my-bucket"
+        }
+    ]
+}`);
+
+const policyWithNoStatements = pulumi.jsonParse(jsonIAMPolicy).apply(policy => {
+    // delete the policy statements
+    policy.Statement = [];
+    return policy;
+});
+```
+
+For more details [view the NodeJS documentation](/docs/reference/pkg/nodejs/pulumi/pulumi/).
+
+{{% /choosable %}}
+{{% choosable language typescript %}}
+
+```typescript
+const jsonIAMPolicy = pulumi.output(`{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Sid": "VisualEditor0",
+            "Effect": "Allow",
+            "Action": [
+                "s3:ListAllMyBuckets",
+                "s3:GetBucketLocation"
+            ],
+            "Resource": "*"
+        },
+        {
+            "Sid": "VisualEditor1",
+            "Effect": "Allow",
+            "Action": "s3:*",
+            "Resource": "arn:aws:s3:::my-bucket"
+        }
+    ]
+}`);
+
+const policyWithNoStatements = pulumi.jsonParse(jsonIAMPolicy).apply(policy => {
```

Comment:
I think the main value being demonstrated here is that we have an `object` instead of a `string`.
Do you think adding type annotations here would help?

```typescript
const iamPolicy: Output<object> = pulumi.jsonParse(jsonIAMPolicy);
const policyWithNoStatements ...
```

(I'm not sure if "object" above is correct, FWIW.)